linetools/analysis/emline.py

```diff
@@ -0,0 +1,63 @@
+""" Utlities for the analysis of emission lines
+"""
+from __future__ import print_function, absolute_import, division, unicode_literals
+
+import numpy as np
+import pdb
+import warnings
+
+from astropy import units as u
+from astropy import constants as const
+from astropy.io import ascii
+from astropy.utils import isiterable
+from linetools.lists.linelist import LineList
+
+# Atomic constant
+atom_cst = (const.m_e.cgs*const.c.cgs / (np.pi * 
+    (const.e.esu**2).cgs)).to(u.AA*u.s/(u.km*u.cm**2))
+
+# e2/(me*c) in CGS
+e2_me_c_cgs = (const.e.esu**2 / (const.c.to('cm/s') * const.m_e.to('g')))
+
+
+# Estimate metallicity from a tried-and-true method
+def metallicity(method, emsystem):
+    """ Calculate a nebular abundance from an input set of Emission lines
+
+    Parameters
+    ----------
+    method : str
+      Name of the method to be applied
+      PG16 --
+          Requires Hbeta, [OII], [OIII], [NII], [SII]
+          Return r_val and s_val
+    emsystem : EmSystem
+
+    Returns
+    -------
+    """
+    if method == 'PG16':
```

Comment:
Can you add a http link to the paper here?

Reply:
This has been added